### DISCUSSION TITLE
Explicit Examples for fulfilling Basic Order

### DISCUSSION THREAD

**Discussion by 0xAskar** [ProjectOpenSea/seaport#571] [Q&A]
Are there any non-confusing explicit examples or tutorials on how to create and fulfill a basic order in javascript using the Seaport protocol? All I've found is some git links with confusing code with some tests they have used and higher level descriptions, but not explicit examples that help people here learning how to use the SDK such as fullfillBasicOrder and fulfillOrder in here https://docs.opensea.io/v2.0/reference/seaport-interface

**Comment by ryanio** [comment]
Yes please see Seaport.js: https://github.com/ProjectOpenSea/seaport-js/blob/main/README.md#getting-started

**Reply by 0xAskar** [reply]
Yes, thanks. I was hoping to see a more explicit example for an exact way to buy an _already listed_ NFT for any using via Metamask. If you have any documentation on that, that would be great. Many people seem to be stuck on problems similar to this (through my time on Opensea Discord), so more detailed tutorials would be great. Thanks!

**Comment by JasperAlexander** [comment]
> Are there any non-confusing explicit examples or tutorials on how to create and fulfill a basic order in javascript using the Seaport protocol? All I've found is some git links with confusing code with some tests they have used and higher level descriptions, but not explicit examples that help people here learning how to use the SDK such as fullfillBasicOrder and fulfillOrder in here https://docs.opensea.io/v2.0/reference/seaport-interface

This is an explicit example of how to use fulfillOrder.

https://github.com/ProjectOpenSea/seaport/discussions/564#discussioncomment-3103257

**Reply by 0xAskar** [reply]
Update:1
Hey, so I realized my problem was that I was using "0x0" instead of "0x0000000000000000000000000000000000000000" for the token address for Ethereum. However, I have a few questions:

Update2:
- Where do you import "setCurrentOrder(executedOrder)" from? Update: I just create a different order variable and had that set as a global for now. 
- Next question is I'm having problems for permissions despite approving all for the offerer previously? The problem can be seen in the screenshot below. Sometimes I get gas estimation errors. Can be seen in the second pic
- I was originally wondering how would you buy an NFT _already listed_ on the marketplace? Do you have an example in this case? 
- 
Here are the errors in text for those who search on here:
`The requested account and/or method has not been authorized by the user.`
`Error: cannot estimate gas; transaction may fail or may require manual gas limit`- 

Update:3
- I think the above gas error came from trying to fulfill a rinkeby trxn on ethereum. I'm not sure how to create a seaport instance for tesnets. However, I'm also not sure how to do the below problem.
-So, Please correct me if I'm wrong, but if I understand correctly, this creates an order and fulfills it as in the item does not have to be listed but won't be listed anyways, it will just create an order and then fulfill it. But here's the problem, if there is an item listed already on Opensea for example, how can I get that order and then fulfill that without needing the offerer permission (as they already listed it) but instead just need the buyer permission?


<img width="540" alt="image" src="https://user-images.githubusercontent.com/37524379/178537956-47d8b17c-71d0-4f11-a646-049d43df7612.png">
<img width="535" alt="image" src="https://user-images.githubusercontent.com/37524379/178539529-b39c1f0f-b154-4891-bf59-51636a93ccad.png">

Also here's my code thus far. Thank you for help so far.

`   
        import * as Web3 from "web3"
        import {ethers} from "ethers";
        import { solidityKeccak256 } from "ethers/lib/utils";
        import { Seaport, BasicOrderTypes } from "@opensea/seaport-js";
        import { ItemType } from "@opensea/seaport-js/lib/constants.js";

        export async function buyNFT(props) {
        const provider = new ethers.providers.Web3Provider(window.ethereum);

        const seaport = new Seaport(provider);

        const offerer = "0xDF93B5ec8Eb9407F03fF14a2E2B3eE4196fcD8b0"
        let order = null
        const sell = async() => {
            if (typeof seaport !== "undefined") {
                console.log("inside")
                const {executeAllActions} = await seaport.createOrder({
                    offer: [
                        {
                            itemType: ItemType.ERC721,
                            token: "0x1fe3139d6290d58c69fb827857e8b2e01eb4455c",
                            identifier: "0"
                        }
                    ],
                    consideration: [
                        {
                            amount: "100000000000000000",
                            token: "0x0000000000000000000000000000000000000000",
                            recipient: Web3.utils.toChecksumAddress(offerer) //offerer, which is the seller in this case to list it
                        }
                    ]
                })
                const executedOrder = await executeAllActions()
                order = executedOrder
            }
        }

        const buy = async() => {
            if (typeof seaport !== "undefined" && order) {
                const { executeAllActions: executeAllFullfillActions} = await seaport.fulfillOrder({
                    order: order,
                    accountAddress: "0x94DBCE8a42E2F578ad80266863bc9C7d2e04ea6A"
                })

                const transaction = await executeAllFullfillActions()
            }
        }

        async function doAll() {
            await sell()
            await buy()
        }
        doAll()
    }`
